openstack/tools/openstack_regression_tests_runner.sh

```diff
@@ -0,0 +1,143 @@
+if ! [[ -e $IMAGES_PATH/fedora-coreos-35.qcow2 ]]; then
+    wget https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/35.20220424.3.0/x86_64/fedora-coreos-35.20220424.3.0-openstack.x86_64.qcow2.xz -O $IMAGES_PATH/fedora-coreos-35.qcow2.xz
+    (cd $IMAGES_PATH; xz -d fedora-coreos-35.qcow2.xz; )
+fi
+
+# Install dependencies
+which yq &>/dev/null || sudo snap install yq
+
+# Ensure zosci-config checked out and up-to-date
+(
+cd
+if [[ -d zosci-config ]]; then
+    cd zosci-config
+    git checkout master
+    git pull
+else
+    git clone https://github.com/openstack-charmers/zosci-config
+fi
+)
+
+TOOLS_PATH=$(realpath $(dirname $0))/func_test_tools
+CHARM_PATH=$(pwd)
+
+echo "Running regression tests"
+
+source ~/novarc
+export {,TEST_}CIDR_EXT=`openstack subnet show subnet_${OS_USERNAME}-psd-extra -c cidr -f value`
+FIP_MAX=$(ipcalc $CIDR_EXT| awk '$1=="HostMax:" {print $2}')
+FIP_MIN=$(ipcalc $CIDR_EXT| awk '$1=="HostMin:" {print $2}')
+FIP_MIN_ABC=${FIP_MIN%.*}
+FIP_MIN_D=${FIP_MIN##*.}
+FIP_MIN=${FIP_MIN_ABC}.$(($FIP_MIN_D + 64))
+
+CIDR_OAM=`openstack subnet show subnet_${OS_USERNAME}-psd -c cidr -f value`
+OAM_MAX=$(ipcalc $CIDR_OAM| awk '$1=="HostMax:" {print $2}')
+OAM_MIN=$(ipcalc $CIDR_OAM| awk '$1=="HostMin:" {print $2}')
+OAM_MIN_ABC=${OAM_MIN%.*}
+OAM_MAX_D=${OAM_MAX##*.}
+# Picking last two addresses and hoping they dont get used by Neutron.
+export {OS,TEST}_VIP00=${OAM_MIN_ABC}.$(($OAM_MAX_D - 1))
+export {OS,TEST}_VIP01=${OAM_MIN_ABC}.$(($OAM_MAX_D - 2))
+
+# More information on config https://github.com/openstack-charmers/zaza/blob/master/doc/source/runningcharmtests.rst
+export {,TEST_}NET_ID=$(openstack network show net_${OS_USERNAME}-psd-extra -f value -c id)
+export {,TEST_}FIP_RANGE=$FIP_MIN:$FIP_MAX
+export {,TEST_}GATEWAY=$(openstack subnet show subnet_${OS_USERNAME}-psd-extra -c gateway_ip -f value)
+export {,TEST_}NAME_SERVER=91.189.91.131
+export {,TEST_}CIDR_PRIV=192.168.21.0/24
+export {,TEST_}SWIFT_IP=10.140.56.22
+export TEST_MODEL_SETTINGS="image-stream=released;default-series=jammy;test-mode=true;transmit-vendor-metrics=false"
+# We need to set TEST_JUJU3 as well as the constraints file
+# Ref: https://github.com/openstack-charmers/zaza/blob/e96ab098f00951079fccb34bc38d4ae6ebb38606/setup.py#L47
+export TEST_JUJU3=1
+
+# NOTE: this should not be necessary for > juju 2.x but since we still have a need for it we add it in
+export TEST_ZAZA_BUG_LP1987332=1
+
+# Some charms point to an upstream constraints file that installs python-libjuju 2.x so we need to do this to ensure we get 3.x
+export TEST_CONSTRAINTS_FILE=https://raw.githubusercontent.com/openstack-charmers/zaza/master/constraints-juju34.txt
+
+LOGFILE=$(mktemp --suffix=-charm-func-test-results)
+(
+    # Ensure charmed-openstack-tester checked out and up-to-date
+    if [[ -d $HOME/charmed-openstack-tester ]]; then
+        cd $HOME/charmed-openstack-tester
+        git checkout master
+        git pull
+    else
+        cd $HOME
```

Comment:
Why not just `cd`? Just asking as you did that [here](https://github.com/canonical/stsstack-bundles/pull/254/files#diff-df917a4f34332b87c75ba0996575e9c7bec5dce969017f0dfdcf4b20bb9dd0a8R69) ;-)

Reply:
ack